packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -365,26 +397,121 @@ export function createRootHooks<
           onStarted: () => {
             if (!isStopped) {
               optsRef.current.onStarted?.();
+              updateState((prev) => ({
+                ...prev,
+                status: 'pending',
+                error: undefined,
+              }));
             }
           },
           onData: (data) => {
             if (!isStopped) {
-              optsRef.current.onData(data);
+              optsRef.current.onData?.(data);
+              updateState((prev) => ({
+                ...prev,
+                status: 'pending',
+                data,
+                error: undefined,
+              }));
             }
           },
-          onError: (err) => {
+          onError: (error) => {
```

Comment:
So something I didn't consider during our discussion the other day

A real downside of handling .reset() from here is the main use cases we have are for cross-cutting concerns like auth expiring and needing a refresh

Putting those in the link is in my opinion much better because that's where auth gets set up in the first place.

In that way I don't really have a test I can write for this scenario because I think we should formalise the global handler on the link

Reply:
Would subscription default options - the way RQ does query defaults etc - be an option?

Reply:
I suppose so but you'd have to consider how to get a handle to the subscription globally, which wouldn't exactly be like RQ as the definition would be different on the global config

Are you concerned about the complexity of keeping it on the link or something else?